src/workers/notifications/utils.ts

```diff
@@ -44,54 +44,94 @@ type GetMembersParams = {
   where: ObjectLiteral;
 };
 
+const filterUsersWithGlobalPreferences = async (
+  con: DataSource,
+  userIds: string[],
+  notificationType: NotificationType,
+  channel: 'email' | 'inApp',
+): Promise<string[]> => {
+  if (!userIds.length) {
+    return [];
+  }
+
+  const users = await con.getRepository(User).find({
+    select: ['id', 'notificationFlags'],
+    where: { id: In(userIds) },
+  });
```

Comment:
I think I commented before that this we could get my join over source member, no need for another query on user table?